### PR TITLE
fix: set yaml read explictly, refactor: various lints

### PR DESCRIPTION
**Pull request by debendraoli**
### Commit Message

```bash
type: commit message
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: test/e2e/conformance_test.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ import (
 
 const (
 	relayerImageEnv    = "RELAYER_IMAGE"
-	relayerImage       = "strangeloveventures/relayer"
+	relayerImage       = "ghcr.io/strangelove-ventures/relayer"
```
Comment (AntonAndell) [inline test/e2e/conformance_test.go:21]: Perhaps unrelated for this PR but why are we using the strangelove relayer? guess it is the only one that has live docker images but will never work for this test

**Review by AntonAndell — APPROVED** [review]

File: test/e2e/conformance_test.go (hunk 1 above)
Comment (debendraoli) [reply test/e2e/conformance_test.go:21]: Oh yeah, I left that unattendaed. I am overrding using env. I think its better to have somewhere like dockerhub.

I have working image pushed here. debendraoli/relayer.

Comment (AntonAndell) [reply test/e2e/conformance_test.go:21]: Yeah we probably should build one whenever ibc-relay gets a push to main in the future, but your own might be better as a default for now if you have it in a public dockerhub

**Review by redlarva — APPROVED** [review]
looks good

**Pull request merged**